django/shortcuts.py

```diff
@@ -61,25 +58,15 @@ def redirect(to, *args, **kwargs):
 
 def _get_queryset(klass):
     """
-    Returns a QuerySet from a Model, Manager, or QuerySet. Created to make
-    get_object_or_404 and get_list_or_404 more DRY.
-
-    Raises a ValueError if klass is not a Model, Manager, or QuerySet.
+    Returns a QuerySet or a Manager. Created to make get_object_or_404 and
+    get_list_or_404 more DRY.
     """
-    if isinstance(klass, QuerySet):
-        return klass
-    elif isinstance(klass, Manager):
-        manager = klass
-    elif isinstance(klass, ModelBase):
-        manager = klass._default_manager
-    else:
-        if isinstance(klass, type):
-            klass__name = klass.__name__
-        else:
-            klass__name = klass.__class__.__name__
-        raise ValueError("Object is of type '%s', but must be a Django Model, "
-                         "Manager, or QuerySet" % klass__name)
-    return manager.all()
+    # If it is a model class, or anything else with ._default_manager
+    if hasattr(klass, '_default_manager'):
+        return klass._default_manager.all()
+    # If it is a manager or queryset, then this is OK. Even if it isn't then
```

Comment:
This would probably be better in the docstring. Maybe you want to say something about ducktyping instead of just "let's hope it has .get()" (guess you might want to add "filter()" too)